test/customer.js

```diff
@@ -65,66 +80,71 @@ describe("Class customer", function(){
-          cust.getPaymentList().then(function (paymentList2) {
-            paymentList2.length.should.equal(2);
-            done();
+    customer.Customer.create("test@email.com","David","Pate").then(function (cust) {
+      custCleanList.push(cust.stripeCusid);
+      var promises = [];
+      var promises2 = [];
+      promises.push(cust.addPayment(sourceData,"tok_visa").catch(done));
+      promises.push(cust.addPayment(sourceData,"tok_mastercard").catch(done));
+      promises.push(cust.addPayment(sourceData,"tok_mastercard").catch(done));
+
+      Promise.all(promises).then(function () {
+        cust.getPaymentList().then(function (paymentList1) {
+          cust.removePayment(paymentList1[0].id).then(function () {
+            cust.getPaymentList().then(function (paymentList2) {
+              paymentList2.length.should.equal(2);
+              done();
+            }).catch(done);
           }).catch(done);
         }).catch(done);
       }).catch(done);
     }).catch(done);
   });
 
   it("Change customer's source", function(done) {
-    var cust = new customer.Customer(JSON.stringify(jsonCust1));
-
-    var actualSource = "";
-    var promises = [];
-
-    promises.push(cust.addPayment(sourceData,"tok_visa").catch(done));
-    promises.push(cust.addPayment(sourceData,"tok_mastercard").catch(done));
-
-    Promise.all(promises).then(function () {
-      stripe.customers.retrieve(jsonCust1.stripeCusid).then(function (custStripe) {
-        actualSource = custStripe.default_source;
-        cust.getPaymentList().then(function (paymentList1) {
-          for (let i in paymentList1) {
-            if (actualSource != paymentList1[i].id) {
-              cust.setStripePayment(paymentList1[i].id).then(function () {
-                stripe.customers.retrieve(jsonCust1.stripeCusid).then(function (custStripe) {
-                  actualSource.should.not.be.equal(custStripe.default_source);
-                  done();
+    customer.Customer.create("test@email.com","David","Pate").then(function (cust) {
+      custCleanList.push(cust.stripeCusid);
+      var actualSource = "";
+      var promises = [];
+
+      promises.push(cust.addPayment(sourceData,"tok_visa").catch(done));
+      promises.push(cust.addPayment(sourceData,"tok_mastercard").catch(done));
+
+      Promise.all(promises).then(function () {
+        stripe.customers.retrieve(cust.stripeCusid).then(function (custStripe) {
+          actualSource = custStripe.default_source;
+          cust.getPaymentList().then(function (paymentList1) {
+
+            for (let i in paymentList1) {
+              if (actualSource != paymentList1[i].id) {
+                cust.setStripePayment(paymentList1[i].id).then(function () {
+                  stripe.customers.retrieve(cust.stripeCusid).then(function (custStripe) {
+                    actualSource.should.not.be.equal(custStripe.default_source);
+                    done();
+                  }).catch(done);
```

Comment:
ici aussi, utilises les `promises` pour éviter le [callback hell](http://stackabuse.com/avoiding-callback-hell-in-node-js/)